typing/jkind.ml

```diff
@@ -990,6 +998,7 @@ module Layout_and_axes = struct
                      TransientTypeOps.equal (Transient_expr.repr ty1)
                        (Transient_expr.repr ty2))
                    seen_args args
+                 && not (context.is_abstract p)
```

Comment:
This one line is the main payload of the patch! And it is fairly subtle to know why this is correct.

I think it would be helpful to have a

```
(* Note [Abstract types in normalization]
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
```

that explains all of this, and then this line can refer to the Note. You can see an example of this pattern at e.g. https://github.com/oxcaml/oxcaml/blob/main/typing/typedecl.ml#L687. This Note style is directly borrowed from GHC.